test/unit_tests/graphql/utils/test_override_class.py

```diff
@@ -0,0 +1,258 @@
+from orchestrator.graphql.utils.override_class import override_class
+from test.unit_tests.config import GRAPHQL_ENDPOINT, GRAPHQL_HEADERS
+
+
+@pytest.fixture
+def override_class_app_graphql(
+    fastapi_app_graphql,
+    test_client,
+    sub_one_subscription_1,
+):
+    def customer_id_override(self) -> str:
+        return "overriden"
+
+    customer_id_override_field = strawberry.field(
+        name="customerId", resolver=customer_id_override, description="Returns customer_id"
+    )
+    customer_id_override_field.name = "customer_id"
+
+    def new(self) -> UUID:
+        return self.customer_id
+
+    new_field = strawberry.field(name="newField", resolver=new, description="Returns new_field")
+    new_field.name = "new_field"
+
+    override_class(CustomerType, {"customer_id": customer_id_override_field, "new_field": new_field})
+
+    fastapi_app_graphql.register_graphql()
+    yield fastapi_app_graphql
+
+    # reset the overriden fields.
+
+    @strawberry.field(description="Returns customer_id")  # type: ignore
+    def customer_id(self) -> str:
+        return self.customer_id
+
+    customer_id.name = "customer_id"
+
+    override_class(CustomerType, {"customer_id": customer_id})
+    CustomerType.__strawberry_definition__._fields = [
+        field for field in CustomerType.__strawberry_definition__._fields if field.name != "new_field"
+    ]
+    fastapi_app_graphql.register_graphql()
+
+
+def get_customers_query(
+    first: int = 1,
+    after: int = 0,
+    filter_by: Union[list[str], None] = None,
+    sort_by: Union[list[dict[str, str]], None] = None,
+    override_fields: bool = False,
+) -> bytes:
+    query = f"""
+query CustomerQuery($first: Int!, $after: Int!, $filterBy: [GraphqlFilter!], $sortBy: [GraphqlSort!]) {{
+  customers(first: $first, after: $after, filterBy: $filterBy, sortBy: $sortBy) {{
+    page {{
+      customerId
+      fullname
+      shortcode
+      { "newField" if override_fields else ""}
+    }}
+  }}
+}}
+    """
+    return json.dumps(
+        {
+            "operationName": "CustomerQuery",
+            "query": query,
+            "variables": {
+                "first": first,
+                "after": after,
+                "sortBy": sort_by if sort_by else [],
+                "filterBy": filter_by if filter_by else [],
+            },
+        }
+    ).encode("utf-8")
+
+
+def get_subscriptions_customer_query(
+    first: int = 1,
+    after: int = 0,
+    filter_by: Union[list[str], None] = None,
+    sort_by: Union[list[dict[str, str]], None] = None,
+    override_fields: bool = False,
+) -> bytes:
+    query = f"""
+query SubscriptionQuery($first: Int!, $after: Int!, $filterBy: [GraphqlFilter!], $sortBy: [GraphqlSort!]) {{
+  subscriptions(first: $first, after: $after, filterBy: $filterBy, sortBy: $sortBy) {{
+    page {{
+      customer {{
+        customerId
+        fullname
+        shortcode
+        { "newField" if override_fields else ""}
+      }}
+    }}
+  }}
+}}
```

Comment:
You could use the old `"foo { %s }" % ("bar", )` formatting to avoid having to escape the curlies